### ISSUE TITLE
Adding stacked series to XYChartScrollbar before it was added to chart results in error

### ISSUE THREAD

**Issue by thunderbolt10** [amcharts/amcharts4#1369]
I create a page with a single stacked line chart and it draws perfectly.
If I then create a second stacked line chart then both charts show the error message "Cannot read property 'stack' of undefined".
If the series 'stacked' flag is set to false then both charts will display correctly.
![image](https://user-images.githubusercontent.com/30114322/59201959-7e7bb280-8b93-11e9-9366-3243d402ab6c.png)

Console shows the following error stack:
`TypeError: Cannot read property 'stack' of undefined
    at LineSeriesDataItem.push../node_modules/@amcharts/amcharts4/.internal/core/DataItem.js.DataItem.setCalculatedValue (DataItem.js:532)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.getStackValue (XYSeries.js:1403)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.processValues (XYSeries.js:863)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/Series.js.Series.rangeChangeUpdate (Series.js:382)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/core/Component.js.Component.validateDataRange (Component.js:556)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.validateDataRange (XYSeries.js:698)
    at DateAxis.js:466
    at Iterator.js:57
    at Iterator.js:10
    at Module.each (Iterator.js:56)`

[https://codepen.io/thunderbolt10/pen/XLrBYJ](https://codepen.io/thunderbolt10/pen/XLrBYJ)
On line 14 change stacked=false and the charts will display in non-stacked format.

**Comment by martynasma** [comment]
This is because of using stacked series in an `XYChartScrollbar`.

First of all you push `series1` into it twice.

Second, stacking calculations require some sort of chart context, so when you add those series to a scrollbar object which is not yet attached to the chart, it gets confused. I know, we should probably handle this situation, but the easiest fix is to just assign the scrollbar to chart before adding any series:

```
  var scrollbarX = new am4charts.XYChartScrollbar();
  chart.scrollbarX = scrollbarX;
  scrollbarX.series.push(series1);
  scrollbarX.series.push(series2);
  scrollbarX.height = am4core.percent(10);
```

**Comment by thunderbolt10** [comment]
Thanks again @martynasma, your spot on and as ever quick to respond.  👍

**Comment by martynasma** [comment]
Fixed in [4.5.0](https://github.com/amcharts/amcharts4/releases/tag/4.5.0).

## [4.5.0] - 2019-06-14

### Added
- Support of queued chart initialization. If enabled (`am4core.options.queue = true`) the chart will not start initializing until previous chart finishes.
- Support for delayed rendering. If enabled (`am4core.options.onlyShowOnViewport = true`) the chart will start rendering itself only when scrolled into view.

### Changed
- Default CSS for a `Popup` now has text color set to ensure it is visible on default background.

### Fixed
- `dataContext` of `ForceDirectedNodeDataItem` was not being populated correctly.
- `MapPolygonSeries` was not being displayed properly if GeoJSON contained a single very small polygon.
- Tooltips now work correctly when a chart is contained within the shadow DOM.
- `rtl` inheritance was not working properly on some elements, e.g. axis tooltips.
- None of the `Responsive` events were being triggered.
- JSON: It was not possible to set `XYSeries.baseAxis` using axis' `id`.
- `LineSeries` was being cut off if div of a chart was more than 20K pixels width/height.
- When axis was disabled (`axis.disabled = true`) its ticks and grid remained visible.
- If a chart had series with bullets and `minBulletDistance` set and the chart was resized so that bullets had to be hidden, some of the bullets remained visible.
- `openCategoryX` and `openCategoryY` data fields were not working properly on `XYSeries`.
- If `XYChart` had stacked series and series had both possitive and negative numbers, series were stacked incorrectly if some of the series were hidden.
- `MapChart` was not adjusting its size to the container in some specific cases.
- Some performance improvements with `series.connect = false` and a lot of missing values.
- `Rectangle3D` was not copying side's properties when cloning.
- `ForceDirectedLink`'s data item was not properly set. It is set to target's data item now.
- `MapChart` was not displaying maps of very small countries properly.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
